src/js/polyfills/webvtt.js

```diff
@@ -551,6 +557,11 @@ function BoxPosition(obj) {
     this.width = obj.width || width;
     this.lineHeight = lh !== undefined ? lh : obj.lineHeight;
 
+    if (cueDiv && div && this.width > div.parentElement.offsetWidth) {
+        cueDiv.style.whiteSpace = 'pre-wrap';
+        div.style.left = '0px';
```

Comment:
Some strange behavior was happening where the caption would get positioned, while extending off screen.
Then the wrap would occur, but the left style would get updated based on the first position, so the caption would still be off screen, but correctly wrapped.
Setting this left style here moved the caption to the correct spot.

Reply:
iirc, `boxposition` is created when we are attempting to find the proper placement for the captions box which is after the `CueStyleBox`. i would like to find a solution where we patch it in CueStyleBox above and then box positions continue to just work ™️  when given any shape of container. 

i know we've gone a bit in circles on this one so i could be misunderstanding the root of the problem but have we tried pulling this logic into the stylle box constructor and seeing what positioning is chosen? is it the same weird behavior we were seeing with the smaller box and/or mispositioning?

Reply:
The only issue we could have with trying to patch it in the CueStyleBox is that the width of the elements are unknown at that point, since they haven't been appended to the page yet. So either the wrapping needs to be determined by something other than the width, or the logic can be performed elsewhere such as processCues?